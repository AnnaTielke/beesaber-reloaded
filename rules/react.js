module.exports = {
  plugins: ['react'],

  extends: ['plugin:react/recommended'],

  settings: {
    react: {
      version: 'detect',
    },
  },

  rules: {
    'react/button-has-type': 'error',
    'react/jsx-boolean-value': 'error',
    'react/jsx-fragments': 'error',
    'react/jsx-handler-names': 'error',
    'react/jsx-no-bind': ['error', { allowArrowFunctions: true }],
    'react/jsx-pascal-case': 'error',
    'react/no-access-state-in-setstate': 'error',
    'react/no-array-index-key': 'error',
    'react/no-danger': 'error',
    'react/no-did-mount-set-state': 'error',
    'react/no-did-update-set-state': 'error',
    'react/no-multi-comp': 'error',
    'react/no-this-in-sfc': 'error',
    'react/no-typos': 'error',
    'react/no-unsafe': 'error',
    'react/no-unused-prop-types': 'error',
    'react/no-unused-state': 'error',
    'react/no-will-update-set-state': 'error',
    'react/prefer-es6-class': 'error',
    'react/prefer-stateless-function': 'error',
    'react/style-prop-object': 'error',
    'react/void-dom-elements-no-children': 'error',
  },
};
