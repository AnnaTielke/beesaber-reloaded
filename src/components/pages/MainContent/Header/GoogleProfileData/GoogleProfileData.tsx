import styled from 'styled-components';

import createLocalImageUrl from '@/helper/createLocalImageUrl';
import { parseAvatarUrl } from '@/helper/urlParser';
import { useAppSelector } from '@/store/hooks';
import { selectUserData } from '@/store/reducer/userDataReducer';

const Container = styled.div`
  max-height: 100px;
`;

const Image = styled.img`
  border-radius 1.5625rem;
  margin-right: 1rem;
  max-width: 3.125rem;
`;

function GoogleProfileData(): JSX.Element | null {
  const userData = useAppSelector(selectUserData);
  const { profilePic, username } = userData;
  const placeholderImage =
    profilePic !== undefined
      ? parseAvatarUrl(profilePic)
      : createLocalImageUrl('m_bee.jpg');

  return (
    <Container>
      <Image
        referrerPolicy="no-referrer" // avoids broken img due to 403 response from google
        src={placeholderImage}
        id="profile-pic-sm"
        alt="profile pic"
      />
      <span>{username}</span>
    </Container>
  );
}

export default GoogleProfileData;
