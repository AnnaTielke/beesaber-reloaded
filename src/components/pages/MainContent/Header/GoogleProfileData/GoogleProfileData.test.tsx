import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import GoogleProfileData from './GoogleProfileData';

import type { UserData } from '@/sharedTypes/UserData';
import type { AppStatus } from '@/store/reducer/appStatusReducer';
import type { RootState } from '@/store/store';

const mockStore = configureMockStore();

const appStatus: AppStatus = {
  isLoggedIn: true,
  isLoggingIn: false,
};

const store: RootState = {
  appStatus,
  notifications: [],
  userData: {} as UserData,
};

describe('GoogleProfileData', () => {
  it('should match the snapshot', () => {
    const { container } = render(
      <Provider store={mockStore(store)}>
        <GoogleProfileData />
      </Provider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});

test.todo('should show fallpack, when no profile picture exists');
test.todo('should show profile picture');
test.todo('match snapshot when logged in');
