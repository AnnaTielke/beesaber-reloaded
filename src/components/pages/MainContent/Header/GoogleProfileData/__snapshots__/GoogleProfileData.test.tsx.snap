// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleProfileData should match the snapshot 1`] = `
.c0 {
  max-height: 100px;
}

.c1 {
  border-radius: 1.5625rem;
  margin-right: 1rem;
  max-width: 3.125rem;
}

<div
  class="c0"
>
  <img
    alt="profile pic"
    class="c1"
    id="profile-pic-sm"
    referrerpolicy="no-referrer"
    src="/public/imgs/m_bee.jpg"
  />
  <span />
</div>
`;
